libs/langchain/langchain/cache.py

```diff
@@ -563,7 +570,15 @@ def update(self, prompt: str, llm_string: str, return_val: RETURN_VAL_TYPE) -> N
             "prompt": prompt,
             "return_val": dumps([g for g in return_val]),
         }
-        llm_cache.add_texts(texts=[prompt], metadatas=[metadata])
+        keys = llm_cache.add_texts(texts=[prompt], metadatas=[metadata])
+
+        # if using TTL, set expiration for the keys that were updated
+        if self.ttl:
+            redis_client = self._cache_dict[self._index_name(llm_string)].client
+            with redis_client.pipeline() as pipe:
```

Comment:
```suggestion
            with redis_client.pipeline(transaction=False) as pipe:
```

Reply:
And we should also add this here:
https://github.com/langchain-ai/langchain/pull/12760/files#diff-65febe4d97a33da425b447d1d4f0212118aa1856a21bea3ca0b4e6321044db39R420